Doc/library/stdtypes.rst

```diff
@@ -3624,13 +3624,15 @@ copying.
 
       .. versionadded:: 3.2
 
-   .. method:: cast(format[, shape])
+   .. method:: cast(format[, shape, readonly])
```

Comment:
The signature here should be `cast(format[, shape[, readonly]])` (note the additional square brackets around the "readonly" parameter, since it's optional).

Reply:
What about something like

cast(format[, shape], *, readonly=`<unchanged>`)

That implies _readonly_ is a keyword-only parameter, which matches your doc string and tests. Using the keyword notation also reduces the bracket nesting.